lib/srv/db/proxyserver.go

```diff
@@ -343,6 +343,9 @@ func (s *ProxyServer) handleConnection(conn net.Conn) error {
 		s.cfg.IngressReporter.ConnectionAuthenticated(ingress.DatabaseTLS, conn)
 		defer s.cfg.IngressReporter.AuthenticatedConnectionClosed(ingress.DatabaseTLS, conn)
 	}
+	if common.EnterpriseDBProtocolValidation(proxyCtx.Identity.RouteToDatabase.Protocol); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
I see this is the only check where enterprise protocol is forbidden with OSS. Should we prevent oracle databases from getting created in db server or auth?

Reply:
Yes It would be nice to detect this as soon as possible. , I have added additional checks preventing Oracle DB being created in OSS build.